src/overmap.cpp

```diff
@@ -2157,7 +2157,11 @@ tripoint overmap::draw_overmap(const tripoint &orig, const draw_data_t &data)
             const std::string new_note = string_input_popup(
                 _("Note (X:TEXT for custom symbol, G; for color):"),
                 45, old_note, color_notes); // 45 char max
-            if(old_note != new_note) {
+            if(new_note == "" && old_note != "") {
```

Comment:
`""` is an empty string. calling `empty()` and comparing with `""` it is completely equivalent. And string_input should probably returns `std::string()`.

To avoid any potential confusing:
```C++
std::string a = "";
std::string b;
// this is always true:
assert(a == b);
// but this *might* not be true:
assert("" == "")
```
In other words: an empty string is the same as a default constructed string.